### PR TITLE
feat: Implemented Go SDK env var handling

### PR DESCRIPTION
**Pull request by jeremytchang**
Also added unit tests for it that will leave existing environment variables undisturbed. (Will store env vars before test and restore env vars after test)

### CODE REVIEW

File: go/rtl/settings.go [hunk 1]
```diff
@@ -37,3 +44,32 @@ func NewSettingsFromFile(file string, section *string) (ApiSettings, error) {
...
+
+	if v, present  := os.LookupEnv(baseUrlEnvKey); present {
+		settings.BaseUrl = v
+	}
+	if v, present  := os.LookupEnv(apiVersionEnvKey); present {
+		settings.ApiVersion = v
+	}
+	if v, present  := os.LookupEnv(verifySslEnvKey); present {
+		s := strings.ToLower(v)
+		settings.VerifySsl = s == "true" || s == "t" || s == "1" || s == "y" || s == "yes"
```
Comment (jkaster) [inline go/rtl/settings.go:59]: fwiw other SDKs have either an `isBool()` or `defaultBool()` type of helper function. In this case, I think we want to default `settings.VerifySSL` to `true`

**Review by jkaster — APPROVED** [review]
Looks good. Do you want to update the GoLook readme for this now, or in another PR?

File: go/rtl/settings.go (hunk 1 above)
Comment (jeremytchang) [reply go/rtl/settings.go:59]: Oh good catch! I will fix.

Comment (jeremytchang) [reply go/rtl/settings.go:59]: I just reread the spec. My logic is correct. It already has setting default to true. Only if it is set does it change.
```
true, t, yes, y, or 1 (case insensitive) to enable SSL verification. Any other value is treated as false. Defaults to true if not set.
```

**Pull request merged**